realm/realm-library/src/objectServer/java/io/realm/ClientResetRequiredError.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import java.io.File;
+
+/**
+ * Class encapsulating information needed for handling a Client Reset event.
+ *
+ * @see io.realm.SyncSession.ErrorHandler#onClientReset(SyncSession, ClientResetRequiredError) for more information
+ *      about when and why Client Reset occur and how to deal with it.
+ */
+public class ClientResetRequiredError extends ObjectServerError {
+
+    private final RealmConfiguration configuration;
+    private final File backupFile;
+    private final File originalFile;
+
+    public ClientResetRequiredError(ErrorCode errorCode, String errorMessage, String backupFilePath, RealmConfiguration configuration) {
+        super(errorCode, errorMessage);
+        this.configuration = configuration;
+        this.backupFile = new File(backupFilePath);
+        this.originalFile = new File(configuration.getPath());
+    }
+
+    /**
+     * Calling this method will execute the Client Reset manually instead of waiting until next app restart. This will
+     * only be possible if all instances of that Realm has been closed, otherwise a {@link IllegalStateException} will
+     * be thrown.
+     *
+     * After this method returns, the backup file can be found in the location returned by {@link #getBackupFile()}.
+     * The file at {@link #getOriginalFile()} have been deleted, but will be recreated from scratch next time a
```

Comment:
I think `have been` is correct here since I'm talking about something that already happened